packages/@o3r/rules-engine/src/services/runner/rules-engine.runner.service.ts

```diff
@@ -73,17 +71,13 @@ export class RulesEngineRunnerService implements OnDestroy {
 
     this.upsertOperators(operatorList);
 
-    this.subscription.add(
-      this.store.pipe(
-        select(selectAllRulesets)
-      ).subscribe((rulesets: Ruleset[]) => this.engine.upsertRulesets(rulesets))
-    );
-
-    this.subscription.add(
-      this.events$.pipe(filter(() => this.enabled)).subscribe(async (events) => {
-        await this.executeActions(events);
-      })
-    );
+    this.store.pipe(
+      select(selectAllRulesets),
+      takeUntilDestroyed()
+    ).subscribe((rulesets: Ruleset[]) => this.engine.upsertRulesets(rulesets));
+    this.events$.pipe(takeUntilDestroyed(), filter(() => this.enabled)).subscribe(async (events) => {
+      await this.executeActions(events);
```

Comment:
It will have to be 
```suggestion
    this.events$.pipe(takeUntilDestroyed(), filter(() => this.enabled)).subscribe((events) => {
      void this.executeActions(events);
```
But I don't get what is simpler 
It's a part of the code I didn't touch